### PR TITLE
Luizoamorim/setup commtizen commitlint

### PR DESCRIPTION
**Pull request by luizoamorim**
### This PR add the commitlint and commitzen setup

These tools will allow us to follow the:
[https://www.conventionalcommits.org/en/v1.0.0/](https://www.conventionalcommits.org/en/v1.0.0/)

You can see the tool documentation here: [https://github.com/commitizen/cz-cli](https://github.com/commitizen/cz-cli)

We're using [husky](https://typicode.github.io/husky/#/?id=articles) which is a tool to help in the git hooks management. 
This way we're setting up a prepare-commit-msg hook to run the commitzen tool before the commit.

Basically this configuration will show an interface to help the developer follow the conventional commits standard:
image

We're also using the commitlint in commit-msg hook. This tool will check if the commit is following the conventional commits standard. But with the commitzen is almost in possible, or impossible don't follow the pattern.

### CODE REVIEW

**Review by imagobea — APPROVED** [review]
Hi Luiz, many thanks for this!

**Can I just ask you to document the expected behaviour?**
Here on the PR description is fine. Eg:

> 1. When typing git commit a dialog will appear on the terminal
> 2. You are meant to use the standard tags

Etc. Thanks again :pray:

File: package.json [hunk 1]
```diff
@@ -45,6 +49,7 @@
   "main": "index.mjs",
   "scripts": {
     "lint": "eslint . --ext .ts",
+    "pre-commit": "git cz",
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply package.json]: I forgot to remove this. Removed.

File: .husky/prepare-commit-msg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+exec < /dev/tty && git cz --hook || true
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply .husky/prepare-commit-msg:4]: Added.

**Review by imagobea — APPROVED** [review]
Thank you @luizoamorim for adding those notes to the PR description. Approved (again), please merge when you can!! :rocket: Cheers
